crates/sui-core/src/consensus_adapter.rs

```diff
@@ -390,9 +404,13 @@ impl ConsensusAdapter {
         &self,
         committee: &Committee,
         tx_digest: &TransactionDigest,
+        amplification_factor: usize,
     ) -> (Duration, usize, usize, usize) {
-        let (position, positions_moved, preceding_disconnected) =
+        let (mut position, positions_moved, preceding_disconnected) =
             self.submission_position(committee, tx_digest);
+        if position < amplification_factor {
+            position = 0;
```

Comment:
I wonder if it is better to use `position = (position + 1).saturating_sub(amplification_factor)` here. It can amplify bandwidth usage even more, but it bounds the latency better which makes paying higher gas worthwhile.

Reply:
Makes sense!